api/server/secrets.go

```diff
@@ -0,0 +1,125 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/libopenstorage/openstorage/secrets"
+)
+
+const (
+	secretKeyOkMsg   = "Secret Key set successfully."
+	secretLoginOkMsg = "Secrets Login Succeeded"
+	secretLoginCheck = "Secrets Login Check Succeeded"
+)
+
+//TODO: Add swagger yaml
+func (c *clusterApi) setClusterSecretKey(w http.ResponseWriter, r *http.Request) {
+
+	params := r.URL.Query()
+	method := "setClusterSecretKey"
+	secretKey := params["clustersecretkey"][0]
+	override, _ := strconv.ParseBool(params["override"][0])
+
+	if secretKey == "" {
+		c.sendError(c.name, method, w, "Missing cluster key", http.StatusInternalServerError)
+		return
+	}
+
+	err := c.SecretManager.Secret.SetClusterSecretKey(secretKey, override)
+	if err != nil {
+		c.sendError(c.name, method, w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Write([]byte("Cluster" + secretKeyOkMsg + "\n"))
```

Comment:
This should be returned as a JSON packet. Create a SetClusterSecretKeyResponse (see other examples) and use that struct to return the information.

Do this for all calls.